service/app/controllers/TelemetryController.java

```diff
@@ -0,0 +1,62 @@
+package controllers;
+
+import com.google.common.net.HttpHeaders;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.controllers.BaseController;
+import play.libs.F;
+import play.libs.F.Promise;
+import play.libs.Json;
+import play.mvc.Result;
+import util.Constant;
+import util.Message;
+
+/**
+ * Telemetry controller handles Telemetry APIs.
+ *
+ * @author Mahesh Kumar Gangula
+ */
+public class TelemetryController extends BaseController {
+  /**
+   * Save telemetry information. Request body either with content type application/json or content
+   * encoding gzip.
+   *
+   * @return Return a promise for telemetry API result.
+   */
+  public F.Promise<Result> save() {
+    try {
+      String contentTypeHeader = request().getHeader(HttpHeaders.CONTENT_TYPE);
+      String encodingHeader = request().getHeader(HttpHeaders.ACCEPT_ENCODING);
+      Request request = new Request();
+      request.setOperation(Constant.DISPATCH_TELEMETRY_OPERATION_NAME);
+      request.put(Constant.HEADERS, request().headers());
+      if (Constant.APPLICATION_JSON.equalsIgnoreCase(contentTypeHeader)) {
+        ProjectLogger.log(
+            "TelemetryController:save: Received telemetry in JSON format.", LoggerEnum.INFO.name());
+        request.put(Constant.BODY, Json.stringify(request().body().asJson()));

```

Comment:
Does request().body() return a null value if no body is sent? We could change the following here:
If body is null or empty, return an appropriate error response instead of generic exception message.

Reply:
Done.